src/sentry/debug_files/artifact_bundle_indexing.py

```diff
@@ -104,7 +102,11 @@ def mark_bundle_for_flat_file_indexing(
                 flat_file_index = flat_file_indexes.pop(0)
                 # remove duplicates from the DB:
                 if len(flat_file_indexes) > 0:
+                    metrics.incr("artifact_bundle_flat_file_indexing.duplicate_index")
                     ids = [index.id for index in flat_file_indexes]
+                    FlatFileIndexState.objects.filter(flat_file_index_id__in=ids).update(
+                        flat_file_index_id=flat_file_index.id
+                    )
```

Comment:
Are we sure that we want to update this? This doesn't imply that the bundles indexed there are actually inside of the bundle.

Reply:
It would be better to set their indexing state to `NOT_INDEXED` so that we can later repair the index.

Reply:
or actually… since we are deleting the `ArtifactBundleFlatFileIndex` entries, all the referenced `FlatFileIndexState` will be removed as well because of the cascading deletes (is that the default btw?)

Reply:
Yep, we use foreign keys with cascade on delete, thus we are already deleting them, good catch actually.

Reply:
answering my own question:

https://github.com/getsentry/sentry/blob/9e96ee69c81dcaf0563359f79c553c690081e9fe/src/sentry/db/models/fields/foreignkey.py#L14

We have cascading deletes by default, so this will automatically clean up duplicated rows in that case which is great.

Reply:
Yep, that is good, so we can remove the `update` of the `FlatFileIndex`.